tests/js/client/server_parameters/test-export-shard-usage-metrics-bytes-per-shard-per-user-cluster.js

```diff
@@ -0,0 +1,1837 @@
+
+      let c = db._create(cn);
+      try {
+        // must insert first to read something back
+        let docs = [];
+        for (let i = 0; i < 10; ++i) {
+          docs.push({ _key: "test" + i, value: i });
+        }
+        c.insert(docs);
+        
+        for (let i = 0; i < 10; ++i) {
+          c.document("test" + i);
+        }
+        
+        // check if the normal metrics endpoint exports any shard-specific metrics
+        let lines = [];
+        getDBServers().forEach((server) => {
+          let res = request({ method: "GET", url: server.url + "/_admin/metrics" });
+          lines = lines.concat(res.body.split(/\n/).filter((l) => l.match(/^arangodb_collection_requests_bytes_(read|written)_total/)));
+        });
+        assertEqual([], lines);
+
+        // check if the usage-metrics endpoint exports any regular metrics
+        lines = [];
+        getDBServers().forEach((server) => {
+          let res = request({ method: "GET", url: server.url + "/_admin/usage-metrics" });
+          // we look for any metric name starting with "rocksdb_" here as a placeholder
+          lines = lines.concat(res.body.split(/\n/).filter((l) => l.match(/^rocksdb_/)));
+        });
+        assertEqual([], lines);
+      } finally {
+        db._drop(cn);
+      }
+    },
+
+    testNoMetricsJustForCreatingCollection : function () {
+      const cn = getUniqueCollectionName();
+
+      let c = db._create(cn);
+      try {
+        let shards = c.shards();
+        assertEqual(1, shards.length);
+
+        let parsed = getParsedMetrics(db._name(), cn);
+        assertEqual({}, parsed);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testHasMetricsWhenReadingFromCollectionSingleReads : function () {
+      [1, 2].forEach((replicationFactor) => {
+        const cn = getUniqueCollectionName();
+
+        let c = db._create(cn, {replicationFactor});
+        try {
+          let shards = c.shards();
+          assertEqual(1, shards.length);
+
+          // must insert first to read something back
+          const n = 50;
+          let docs = [];
+          for (let i = 0; i < n; ++i) {
+            docs.push({ _key: "test" + i, value: i });
+          }
+          c.insert(docs);
+          
+          for (let i = 0; i < n; ++i) {
+            c.document("test" + i);
+          }
+          
+          let parsed = getParsedMetrics(db._name(), cn);
+          assertTrue(parsed.reads[shards[0]] > n * 40, {parsed, replicationFactor});
+          assertTrue(parsed.reads[shards[0]] < n * 50, {parsed, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor, {parsed, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor, {parsed, replicationFactor});
+        } finally {
+          db._drop(cn);
+        }
+      });
+    },
+    
+    testHasMetricsWhenReadingFromCollectionSingleReadsLarge : function () {
```

Comment:
Could be "merged" with the above test method. As far as I can see only the payload differs here (could be parameterized (minor).